plugins/processors/cumulative_sum/cumulative_sum.go

```diff
@@ -0,0 +1,120 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package cumulative_sum
+
+import (
+	_ "embed"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type CumulativeSum struct {
+	Fields            []string        `toml:"fields"`
+	KeepOriginalField bool            `toml:"keep_original_field"`
+	ResetInterval     config.Duration `toml:"reset_interval"`
+	Log               telegraf.Logger `toml:"-"`
+
+	accept    filter.Filter
+	cache     map[uint64]telegraf.Metric
+	nextReset time.Time
+}
+
+var timeNow = time.Now
+
+func (*CumulativeSum) SampleConfig() string {
+	return sampleConfig
+}
+
+func (c *CumulativeSum) Init() error {
+	if len(c.Fields) == 0 {
+		c.Fields = []string{"*"}
+	}
+	f, err := filter.Compile(c.Fields)
+	if err != nil {
+		return fmt.Errorf("failed to create new field filter: %w", err)
+	}
+	c.accept = f
+
+	c.cache = make(map[uint64]telegraf.Metric)
+
+	if c.ResetInterval > 0 {
+		c.nextReset = timeNow().Add(time.Duration(c.ResetInterval))
+	}
```

Comment:
With the modified `Apply` this should be
```suggestion
	c.cache = make(map[uint64]*entry)
```